taxjar/taxjar-magento2-extension
# Integration tests for partial refunds on bundles

### Context

Adds 2 new integration tests, "testBundledProductsOrder" and "testBundledProductsRefund".

### Description


`testBundledProductsOrder` creates an order with a bundle product with 4 options.  This test ensures the `build` method of Taxjar\SalesTax\Model\Transaction\Order correctly generates the expected line items.

`testBundledProductsRefund` creates a creditmemo with a partial refund of the bundle product (refunding 2 of the 4 items).  This test ensures the `build` method Taxjar\SalesTax\Model\Transaction\Refund correctly generates the expected line items.

Both tests have order/product/refund data included via the @magentoDataFixture tags.


### Performance

This PR increases the amount of time it takes to run integration tests.  

### Testing

1. Follow the "Tests" section in the README but stop before actually running the tests
2.  change to the dev/tests/integration directory, ex: `cd /var/www/magento2/dev/tests/integration`
3. run each specific individually, ex: `../../../vendor/bin/phpunit -c phpunit.xml --testsuite "Taxjar" --filter testBundledProductsOrder`
  *note: tests can take 10 minutes each to run depending on hardware
4. Confirm that all tests run successfully

![Screen Shot 2020-05-06 at 7 07 59 PM](https://user-images.githubusercontent.com/44789510/81243727-e322d580-8fcd-11ea-8c61-7568c852db56.png)

#### Versions
